### PR TITLE
build: fetch cni-plugins from kubernetesartifacts storage

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:
Moves the download source for `cni-plugins` archives to our preferred production blob storage.

This is third step in an effort to move off the `acs-mirror` blob storage (and replace manual steps with automated pipelines).

**Issue Fixed**:
Refs #2545

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/azenvtypes.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ var (
 		EtcdDownloadURLBase:              "https://acs-mirror.azureedge.net/github-coreos",
 		KubeBinariesSASURLBase:           "https://acs-mirror.azureedge.net/wink8s/",
 		WindowsTelemetryGUID:             "fb801154-36b9-41bc-89c2-f4d4f05472b0",
-		CNIPluginsDownloadURL:            "https://acs-mirror.azureedge.net/cni/cni-plugins-amd64-" + CNIPluginVer + ".tgz",
+		CNIPluginsDownloadURL:            "https://kubernetesartifacts.azureedge.net/cni-plugins/" + CNIPluginVer + "/binaries/cni-plugins-amd64-" + CNIPluginVer + ".tgz",
```
Comment (jackfrancis) [inline pkg/api/azenvtypes.go]: I think we want to sync up w/ mooncake config owners to ensure we don't inadvertently make their existing acs-mirror mirror (sorry) stale.

@feiskyer @andyzhangx do you maintain `https://mirror.azk8s.cn`? If so, do you know if it is manually maintained, or does it pull from acs-mirror automatically?

Comment (mboersma) [reply pkg/api/azenvtypes.go]: Ideally, we could have the backend pipeline also populate `mirror.azk8s.cn`. cc: @jchauncey.

Comment (andyzhangx) [reply pkg/api/azenvtypes.go]: @jackfrancis
it's automatically populated from https://github.com/containernetworking/plugins, all binaries are under:
https://mirror.azk8s.cn/kubernetes/containernetworking-plugins/

Comment (jackfrancis) [reply pkg/api/azenvtypes.go]: Thanks @andyzhangx! @mboersma I think this means we are good to go here. As we continue to do this work to migrate off of acs-mirror, let's continue to check in w/ mooncake folks to ensure we don't break those mirrors; sounds like we're in good shape

**Pull request merged**